index/server/pkg/server/endpoint.go

```diff
@@ -543,6 +543,12 @@ func buildIndexAPIResponse(c *gin.Context, indexType string, wantV1Index bool, p
 	if params.Deprecated != nil {
 		util.FilterDevfileDeprecated(&index, *params.Deprecated, wantV1Index)
 	}
+	if index == nil || len(index) == 0 {
+		c.JSON(http.StatusOK, gin.H{
+			"status": "No samples found in the registry",
```

Comment:
Because this response is returned if index has no entries and not just the samples, the status should mention stacks too:
```suggestion
			"status": "No stacks or samples found in the registry",
```